### DISCUSSION TITLE
Is there any video source to implement google auth?

### DISCUSSION THREAD

**Discussion by igmyung** [supabase/supabase#13218] [Questions] [answered]
Hi, i want to implement google authentication with swift-supabase. Since i’m beginner to swift, it’s pretty hard to follow the documentation. Is there any video source that teaches how to implement google authentication?

**Comment by dshukertjr** [comment, marked as answer]
Hi @igmyung 

Thanks for the suggestion for creating a video for implementing Google auth on Swift. Actually, currently we are in the progress of bringing proper third party OAuth to Supabase on native apps. Once this is completed, we will most likely create more materials including videos on how to implement it in mobile apps.

**Reply by igmyung** [reply]
@dshukertjr 
What do you mean by 'bringing proper third party OAuth to Supabase on native apps'? Is it mean that we can do OAuth not Safari way, but pure native way?

Also, is it possible to know when will the update be completed?

**Reply by dshukertjr** [reply]
@igmyung 
> Is it mean that we can do OAuth not Safari way, but pure native way?

Yup!

> Also, is it possible to know when will the update be completed?

Can't promise anything, but keep an eye for the next launch week 👀

**Comment by aaronksaunders** [comment]
@dshukertjr this approach does not work for ionic apps or Cordova based apps because if you don’t set the redirect url properly it just doesn’t work. In fact if you use a web clientid google will error out… so there is no workaround at all
